azure/services/scalesets/scalesets.go

```diff
@@ -144,8 +144,11 @@ func (s *Service) Reconcile(ctx context.Context) (retErr error) {
 		}
 	}
 
-	// if we get to here, we have completed any long running VMSS operations (creates / updates)
+	// If we get to here, we have completed any long running VMSS operations (creates / updates)
 	s.Scope.DeleteLongRunningOperationState(s.Scope.ScaleSetSpec().Name, serviceName)
+	// This also means that the VMSS extensions were successfully installed
+	s.Scope.UpdatePutStatus(infrav1.BootstrapSucceededCondition, serviceName, nil)
```

Comment:
@CecileRobertMichon I went ahead and added the UpdatePutStatus() call in the scaleset service. However, I'm not sure how we want to set the error message for VMSSExtensions. Do we want to just set it with whatever error comes from `s.createVMSS(ctx)`, `s.patchVMSSIfNeeded(ctx, fetchedVMSS)`, and `s.getVirtualMachineScaleSetIfDone(ctx, future)`? Or should we try to set specific errors caused by the VMSSExtension (and not necessarily the scaleset as a whole)?

Reply:
That's a good question... looks like right now we're not setting conditions at all for the scale set itself because it hasn't yet been refactored (it was implemented as async before the async proposal landed so it's not fully following the same patterns). I would say let's leave this for now and add a note to handle both VMSS and VMSS extension conditions when an error occurs.